.github/actions/setup/action.yml

```diff
@@ -0,0 +1,44 @@
+name: Setup environment
+
+inputs:
+  cache:
+    description: Enable caching
+    default: "true"
+  node:
+    description: The Node.js version to install
+    required: true
+  solana:
+    description: The Solana version to install
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup pnpm
+      uses: pnpm/action-setup@v3
+    - name: Setup Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: ${{ inputs.node }}
+        cache: "pnpm"
+    - name: Install dependencies
+      run: pnpm install --frozen-lockfile
+      shell: bash
+    - name: Set env vars
+      shell: bash
+      run: pnpm ci:set-env
+    - name: Install Rustfmt
```

Comment:
We probably don't need to install `fmt` and `clippy` in every setup, only on the specific steps for formatting and clippy check. So I think we can remove the steps:
* `Set env vars`
* `Install Rustfmt`
* `Install Clippy`

Reply:
I agree. I think maybe what's better is to do away with the setup action, and just add the steps to each job. What do you think?

Reply:
For now I think I'll leave rustfmt and clippy out of this PR, and instead introduce a new one after this merges that contains what we talked about for the new pipeline layout.